Flan Refactor + spellaftereffects

Haevacht opened the pull request:
If too feature-y just close it and say so.

Sets flans up better, splits the water one away from the base, adds the damage resistances.

Incorporates a spellcast timer. Meant to be changed on inheritance, but editable for badmins. Increased slightly, increased slightly again for fire flans.

Also adds the ability to have aftereffects on your spells, such as lighting shit on fire.

Changes the pathing in the dragoon tomb to get the right flans.

Haevacht committed:
Splits the water flan away from the base and adds spellaftereffects

Haevacht committed:
updates the tomb

Haevacht committed:
fixes them dying in spess thanks joan

ChangelingRain commented on code/modules/mob/living/simple_animal/hostile/flan.dm:
```diff
@@ -59,6 +66,26 @@
+
+/mob/living/simple_animal/hostile/flan/fire/spellaftereffects(mob/living/A)
+	A.ExtinguishMob()
+	return
```
trailing return!

ChangelingRain commented on code/modules/mob/living/simple_animal/hostile/flan.dm:
```diff
@@ -59,6 +66,26 @@
+/mob/living/simple_animal/hostile/flan/fire/spellaftereffects(mob/living/A)
+	A.adjust_fire_stacks(2)
+	A.IgniteMob()
+	return
```
traaaailing retuuuuurn

ChangelingRain commented on code/modules/mob/living/simple_animal/hostile/flan.dm:
```diff
@@ -13,42 +13,49 @@
 
 /mob/living/simple_animal/hostile/flan/New()		//Required for the inheritance of casting animations.
 	..()
 	casting = 0
```
this line doesn't do anything? it's already 0 by the time this happens

Haevacht committed:
slowly learning how procs work

now with 100% less trailing returns, fire flans extinguishing and 100%
more comments about varedits
